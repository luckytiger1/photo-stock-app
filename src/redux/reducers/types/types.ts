export type ImageListStateType = {
  images: (string | object)[];
  loading: boolean;
  error: null | Error;
  searchResults: (string | object)[];
  term: string;
  searchHistory: (string | object)[];
};

export type FavoritesListStateType = {
  favorites: object[];
};

export type PageStateTypes = {
  page: number;
};

export type RelatedListStateType = {
  relatedImages: object[];
};

export type TagsStateType = {
  tags: string[];
};

export interface AppState {
  imageList: ImageListStateType;
  pages: PageStateTypes;
  favoritesList: FavoritesListStateType;
  relatedList: RelatedListStateType;
  tagsList: TagsStateType;
}
