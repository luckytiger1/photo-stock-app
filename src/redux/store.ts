import { createStore, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension';
import createSagaMiddleware from 'redux-saga';
import reducer from './reducers/reducers';
import rootSaga from './sagas/sagas';

const saga = createSagaMiddleware();

const middlewares = [];

if (process.env.NODE_ENV === 'development') {
  middlewares.push(logger);
}

const store = createStore(
  reducer,
  composeWithDevTools(applyMiddleware(...middlewares, saga)),
);

saga.run(rootSaga);

export default store;
