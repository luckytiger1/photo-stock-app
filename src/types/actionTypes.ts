import { Action } from 'redux';

export const ADVANCE_PAGE = 'ADVANCE_PAGE';
export const RESET_PAGE = 'RESET_PAGE';
export const FETCH_IMAGES_REQUEST = 'FETCH_IMAGES_REQUEST';
export const FETCH_IMAGES_SUCCESS = 'FETCH_IMAGES_SUCCESS';
export const FETCH_IMAGES_FAILURE = 'FETCH_IMAGES_FAILURE';
export const SET_TERM = 'SET_TERM';
export const ADD_TO_FAVORITES = 'ADD_TO_FAVORITES';
export const FETCH_IMAGES = 'FETCH_IMAGES';
export const SET_RELATED_IMAGES = 'SET_RELATED_IMAGES';
export const GET_RELATED_IMAGES = 'GET_RELATED_IMAGES';
export const SEARCH_IMAGES_SUCCESS = 'SEARCH_IMAGES_SUCCESS';
export const SEARCH_IMAGES = 'SEARCH_IMAGES';
export const SET_TAGS = 'SET_TAGS';
export const GET_TAGS = 'GET_TAGS';

interface AdvancePageAction extends Action {
  type: typeof ADVANCE_PAGE;
}

interface ResetPageAction extends Action {
  type: typeof RESET_PAGE;
}

interface ImagesRequestedAction extends Action {
  type: typeof FETCH_IMAGES_REQUEST;
}
interface ImagesLoadedAction extends Action {
  type: typeof FETCH_IMAGES_SUCCESS;
  payload: object[];
}
interface ImagesFailedAction extends Action {
  type: typeof FETCH_IMAGES_FAILURE;
  payload: Error;
}
interface SetTermAction extends Action {
  type: typeof SET_TERM;
  payload: string;
}
interface AddToFavoritesAction extends Action {
  type: typeof ADD_TO_FAVORITES;
  payload: object;
}
interface FetchImagesAction extends Action {
  type: typeof FETCH_IMAGES;
}
interface SetRelatedImagesAction extends Action {
  type: typeof SET_RELATED_IMAGES;
  payload: object[];
}
interface GetRelatedImagesAction extends Action {
  type: typeof GET_RELATED_IMAGES;
  payload: object[];
}
interface SearchImagesLoadedAction extends Action {
  type: typeof SEARCH_IMAGES_SUCCESS;
  payload: object[];
}
interface SearchImagesAction extends Action {
  type: typeof SEARCH_IMAGES;
}
interface SetTagsAction extends Action {
  type: typeof SET_TAGS;
  payload: string[];
}
interface GetTagsAction extends Action {
  type: typeof GET_TAGS;
  payload: string;
}

export type PagesActionTypes = AdvancePageAction | ResetPageAction;
export type ImagesActionTypes =
  | ImagesRequestedAction
  | ImagesLoadedAction
  | ImagesFailedAction
  | SetTermAction
  | AddToFavoritesAction
  | FetchImagesAction;
export type RelatedImagesActionTypes =
  | SetRelatedImagesAction
  | GetRelatedImagesAction;

export type SearchImagesActionTypes =
  | SearchImagesLoadedAction
  | SearchImagesAction;

export type TagsActionTypes = SetTagsAction | GetTagsAction;

export type AppActions =
  | PagesActionTypes
  | ImagesActionTypes
  | RelatedImagesActionTypes
  | SearchImagesActionTypes
  | TagsActionTypes;
