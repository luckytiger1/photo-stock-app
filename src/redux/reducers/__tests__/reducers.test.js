import { createStore } from 'redux';
import rootReducer from '../reducers';
import { ADVANCE_PAGE } from '../../../types/actionTypes';

const store = createStore(rootReducer);

describe('rootReducer', () => {
  const action = { type: ADVANCE_PAGE };
  it('should initialize correctly', () => {
    expect(store.getState()).toMatchSnapshot();
    store.dispatch(action);
    expect(store.getState().pages).toEqual({ page: 2 });
  });
});
