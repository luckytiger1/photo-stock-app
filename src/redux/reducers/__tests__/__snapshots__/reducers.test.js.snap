// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rootReducer should initialize correctly 1`] = `
Object {
  "favoritesList": Object {
    "favorites": Array [],
  },
  "imageList": Object {
    "error": null,
    "images": Array [],
    "loading": true,
    "searchHistory": Array [],
    "searchResults": Array [],
    "term": "",
  },
  "pages": Object {
    "page": 1,
  },
  "relatedList": Object {
    "relatedImages": Array [],
  },
  "tagsList": Object {
    "tags": Array [],
  },
}
`;
